### ISSUE TITLE
NullReferenceException in CollectibleBehaviorArtPigment.OnUnloaded

### ISSUE THREAD

**Issue by sirStroam** [anegostudios/VintageStory-Issues#1056]
**Game Version:** 1.15 pre 6
**Platform:** Windows
**Modded:** No

### Description
Loaded world
https://discord.com/channels/302152934249070593/334233095374897152/857737238108045342

### How to reproduce
1. Make a '....'
2. Use it on '....'
3. It crashes / doesn't do the thing.

### Expected behavior
If not obvious, what did you expect to happen?

### Screenshots
Attach a screenshot if it helps explain the issue / shows a visual bug.

### Logs
```
Running on 64 bit Windows with 8 GB RAM 
Version: v1.15.0-pre.6 (Unstable)6/24/2021 4:37:37 PM: Critical error occurred
System.NullReferenceException: Object reference not set to an instance of an object.
   at Vintagestory.GameContent.CollectibleBehaviorArtPigment.OnUnloaded(ICoreAPI api)
   at Vintagestory.API.Common.CollectibleObject.OnUnloaded(ICoreAPI api)
   at Vintagestory.Client.NoObf.ClientMain.Dispose()
   at Vintagestory.Client.NoObf.ClientMain.DestroyGameSession(Boolean gotDisconnected)
   at _cCzRpONKdXthi9PFjAKWxJaCGSc._zAMuaneFGa4jO7fVDia9cefIpm6()
   at Vintagestory.API.Client.GuiElementTextButton.OnMouseUpOnElement(ICoreClientAPI api, MouseEvent args)
   at Vintagestory.API.Client.GuiElement.OnMouseUp(ICoreClientAPI api, MouseEvent args)
   at Vintagestory.API.Client.GuiElementTextButton.OnMouseUp(ICoreClientAPI api, MouseEvent args)
   at Vintagestory.API.Client.GuiComposer.OnMouseUp(MouseEvent mouse)
   at _dH3hMDelu5YQP777ab2Z3IUY6g._NkW36gdk1xxNg5ly7EdIsHPDiOH(MouseEvent )
   at Vintagestory.Client.NoObf.ClientPlatformWindows.Mouse_ButtonUp(Object sender, MouseButtonEventArgs e)
   at System.EventHandler`1.Invoke(Object sender, TEventArgs e)
   at System.EventHandler`1.Invoke(Object sender, TEventArgs e)
   at OpenTK.Platform.Windows.WinGLNative.WindowProcedure(IntPtr handle, WindowMessage message, IntPtr wParam, IntPtr lParam)
   at OpenTK.Platform.Windows.Functions.DispatchMessage(MSG& msg)
   at OpenTK.Platform.Windows.WinGLNative.ProcessEvents()
   at OpenTK.GameWindow.Run(Double updates_per_second, Double frames_per_second)
   at _Bm4NBvFGhkYYb6kZRXCyCXFHRob._0ZviBJOkkLsgH2nWPiNX4k45MmO(_WXS5zDsN00P20IS9kaePFXF9aGe , String[] )
   at _oFmKxmdL6yMMXIdkAAa9Rh1eAhj._0ZviBJOkkLsgH2nWPiNX4k45MmO(ThreadStart )
-------------------------------

Event Log entries containing Vintagestory.exe, the latest 3
==================================
{ TimeGenerated = 6/20/2021 4:47:40 PM, Site = , Source = Application Hang, Message = The program Vintagestory.exe version 1.15.0.0 stopped interacting with Windows and was closed. To see if more information about the problem is available, check the problem history in the Security and Maintenance control panel.

Process ID: 2e58

Start Time: 01d7661b7d8da3ae

Termination Time: 4294967295

Application Path: C:\Users\savag\AppData\Roaming\Vintagestory\Vintagestory.exe

Report Id: c5b41dd0-442e-4e5c-af52-b9fe41da8b1e

Faulting package full name: 

Faulting package-relative application ID: 

Hang type: Top level window is idle
 }
--------------
{ TimeGenerated = 6/19/2021 7:19:54 PM, Site = , Source = Windows Error Reporting, Message = Fault bucket 1259147309639579605, type 5
Event Name: RADAR_PRE_LEAK_64
Response: Not available
Cab Id: 0

Problem signature:
P1: Vintagestory.exe
P2: 1.15.0.0
P3: 10.0.19041.2.0.0
P4: 
P5: 
P6: 
P7: 
P8: 
P9: 
P10: 

Attached files:
\\?\C:\Users\savag\AppData\Local\Temp\RDR3D8B.tmp\empty.txt
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER3DAC.tmp.WERInternalMetadata.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER3DCC.tmp.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER3E63.tmp.csv
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER3EB2.tmp.txt

These files may be available here:


Analysis symbol: 
Rechecking for solution: 0
Report Id: 4a242a70-58ba-4a38-8415-f439a8fed0cf
Report Status: 268435456
Hashed bucket: c9b9d3809ce1ec04417963ce8ef4b7d5
Cab Guid: 0 }
```

**Comment by Craluminum2413** [comment]
Fixed in 1.15.0-pre.8

**Comment by copygirl** [comment]
> Fixed: Might fix an crash odd bug in BehaviorArtPigment

~ [1.15.0-pre.8](https://info.vintagestory.at/v1dot15#pre8)

Please comment if this wasn't fixed.
